subgraphs/playback/src/utils/mocks.ts

```diff
@@ -1,5 +1,190 @@
-export const mocks = {
-  User: () => ({
-    displayName: 'Mocked Watson'
-  })
-}
\ No newline at end of file
+import { Spotify } from '../dataSources/spotify.types';
+import { SpotifyDataSource } from '../dataSources/spotify';
+
+const userState: {
+  [userId: string]: {
+    actions: { disallows: [string] };
+    context: {
+      external_urls: { spotify: string };
+      href: string;
+      type: 'artist' | 'playlist' | 'album' | 'show';
+      uri: string;
+    };
+    device: {
+      id: string;
+      is_active: boolean;
+      is_private_session: boolean;
+      is_restricted: boolean;
+      name: string;
+      type: string;
+      volume_percent: number;
+    };
+    is_playing: boolean;
+    item: { id: string };
+    progress_ms: number;
+    repeat_state: 'context' | 'off' | 'track';
+    shuffle_state: boolean;
+    timestamp: number;
+    currently_playing_type: string;
+  };
+} = {
+  default: {
+    actions: { disallows: ['INTERRUPTING_PLAYBACK'] },
+    context: {
+      external_urls: { spotify: '' },
+      href: '',
+      type: 'album',
+      uri: '',
+    },
+    device: {
+      id: 'f15b1cd24a5ae5fe4224edc5d4958a06f07c5b99',
+      is_private_session: false,
+      is_restricted: false,
+      name: `My Computer`,
+      type: 'Computer',
+      is_active: true,
+      volume_percent: 100,
+    },
+    is_playing: true,
+    item: { id: '2jJIENqpTOjBECelzBJAVL' },
+    progress_ms: 0,
+    repeat_state: 'off',
+    shuffle_state: false,
+    timestamp: Date.now(),
+    currently_playing_type: 'track',
+  },
+};
+
+export function addUser(userId: string) {
+  console.log(`trying to add user ${userId}`);
+  if (userState[userId]) return;
+
+  //TODO - pick a song to use for playback state
+  userState[userId] = userState['default'];
+}
+
+export class MockedSpotifyDataSource implements SpotifyDataSource {
+  private userId: string;
+  constructor(userId: string) {
+    this.userId = userId;
+  }
+  getDevices(): Promise<Spotify.Object.List<'devices', Spotify.Object.Device>> {
```

Comment:
```suggestion
  getDevices() {
```

Since this class implements the `SpotifyDataSource` interface, do you need the explicit return values here?

Reply:
Found out that this is not actually possible in TypeScript: https://github.com/microsoft/TypeScript/pull/6118#issuecomment-216595207. No need to change anything here.